packages/blade/src/components/Box/BaseBox/useMemoizedStyles.web.ts

```diff
@@ -9,7 +9,7 @@ const getDependencyProps = (props: BaseBoxProps & { theme?: Theme }): string | B
   const { theme, children, className, id, ...rest } = props;
   let dependencyPropString: string | BaseBoxProps = '';
   try {
-    dependencyPropString = JSON.stringify(rest);
+    dependencyPropString = `${JSON.stringify(rest)}-${theme?.name}`;
```

Comment:
So we need to update the memoization of our Box component whenever something is changed in the theme for example spacing tokens, right?

Reply:
With this approach, This won't exactly get triggered if someone takes the object and changes something manually in between in the state (I hope no one does that 🙈). It will only work when someone switches the themeToken itself.